openedx/features/survey_report/static/survey_report/js/admin_banner.js

```diff
@@ -4,8 +4,26 @@ $(document).ready(function(){
       // If you want to do something after the slide-up, do it here.
       // For example, you can hide the entire div:
       // $(this).hide();
+      var userId = document.getElementById('userIdSurvey').value;
+      // Store the dismissal time in milliseconds
+      var dismissalTime = new Date().getTime();
+      // Calculate the expiration time (1 month in milliseconds)
+      var expirationTime = dismissalTime + (30 * 24 * 60 * 60 * 1000); // 30 days
+      // Store the dismissal time in the browser's local storage
+      localStorage.setItem('bannerDismissalTime_' + userId, dismissalTime);
+      localStorage.setItem('bannerExpirationTime_' + userId, expirationTime);
     });
   });
+
+    // Check if the banner should be shown or hidden on page load
+  var userId = document.getElementById('userIdSurvey').value;
+  var bannerDismissalTime = localStorage.getItem('bannerDismissalTime_' + userId);
+  var bannerExpirationTime = localStorage.getItem('bannerExpirationTime_' + userId);
```

Comment:
What would happen if we access from incognito mode?

Reply:
If we access from incognito, the localstorage woulb be empty. But as long as the incognito tab remains open the local storage for that tab should not be erased. Is there a lot of users that navigate using incognito? DO you think it might be a problem?

Reply:
I understand. From my side I think there is no problem with it working that way.